gfx/common/x11_common.c

```diff
@@ -147,6 +153,75 @@ void x11_set_window_attr(Display *dpy, Window win)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
+    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+#ifdef HAVE_DBUS
+    x11_suspend_screensaver_dbus(enable);
+#endif
+}
+
+#ifdef HAVE_DBUS
+void x11_suspend_screensaver_dbus(bool enable)
+{
+    const char *app = "RetroArch";
+    const char *reason = "Playing a game";
+    DBusMessage *msg, *reply;
+
+    if (!enable)
+       return;
+
+    if (dbus_screensaver_cookie > 0)
+        return; // Already suspended
+
+    if (dbus_connection == NULL)
+        return; // DBus connection was not obtained
+
+    
+    msg = dbus_message_new_method_call("org.freedesktop.ScreenSaver",
```

Comment:
Not here. x11_window_destroy doesn't check if dbus_screensaver_cookie is nonzero.